### PR TITLE
SAR-9483 Update audit for identifiers match

### PR DESCRIPTION
**Pull request by woollett-r**

### CODE REVIEW

File: app/uk/gov/hmrc/partnershipidentificationfrontend/models/PartnershipFullJourneyData.scala [hunk 1]
```diff
@@ -49,11 +54,29 @@ object PartnershipFullJourneyData {
...
-      (JsPath \ IdentifiersMatchKey).write[Boolean] and
+      (JsPath \ IdentifiersMatchKey).write[ValidationResponse] and
       (JsPath \ BusinessVerificationKey).writeNullable[BusinessVerificationStatus] and
       (JsPath \ RegistrationStatusKey).write[RegistrationStatus](RegistrationStatus.format)
     ) (unlift(PartnershipFullJourneyData.unapply))
 
   implicit val format: OFormat[PartnershipFullJourneyData] = OFormat(reads, writes)
 
+  val jsonWriterForCallingServices: Writes[PartnershipFullJourneyData] =
+    (partnershipFullJourneyData: PartnershipFullJourneyData) =>
```
[the remark this answers is not in the document]
Comment (woollett-r) [reply app/uk/gov/hmrc/partnershipidentificationfrontend/models/PartnershipFullJourneyData.scala:65]: Done

[the remark this answers is not in the document]
Comment (woollett-r) [reply app/uk/gov/hmrc/partnershipidentificationfrontend/models/PartnershipFullJourneyData.scala:65]: Done

File: app/uk/gov/hmrc/partnershipidentificationfrontend/models/ValidationResponse.scala [hunk 2]
```diff
@@ -16,10 +16,50 @@
...
+    override def reads(jsValue: JsValue): JsResult[ValidationResponse] = jsValue.validate[String] match {
+      case JsSuccess(validationResponseAsString, _) => validationResponseAsString match {
+        case IdentifiersMatchedKey => JsSuccess(IdentifiersMatched)
+        case IdentifiersMismatchKey => JsSuccess(IdentifiersMismatch)
+        case UnMatchableKey => JsSuccess(UnMatchable)
+      }
+      case JsError(_) =>
+        jsValue.validate[Boolean] match {
+          case JsSuccess(identifiersMatch, _) => if(identifiersMatch) JsSuccess(IdentifiersMatched) else JsSuccess(IdentifiersMismatch)
+          case JsError(_) => throw new InternalServerException(("Invalid validation response"))
```
[the remark this answers is not in the document]
Comment (woollett-r) [reply app/uk/gov/hmrc/partnershipidentificationfrontend/models/ValidationResponse.scala:49]: Done

File: app/uk/gov/hmrc/partnershipidentificationfrontend/service/ValidationOrchestrationService.scala [hunk 3]
```diff
@@ -29,44 +29,47 @@ class ValidationOrchestrationService @Inject()(partnershipIdentificationService:
...
       }
-      case _ =>
-        throw new InternalServerException(s"No data stored for journeyId: $journeyId")
+  }
+
+  private def matchTrustDetails(optSaInformation: Option[SaInformation], optCompanyProfile: Option[CompanyProfile])
+                       (implicit hc: HeaderCarrier): Future[ValidationResponse] = {
+    (optSaInformation, optCompanyProfile) match {
+      case (Some(SaInformation(sautr, postcode)), None) => for {
+        identifiersMatch <- validatePartnershipInformationService.validateIdentifiers(sautr, postcode)
```
[the remark this answers is not in the document]
Comment (woollett-r) [reply app/uk/gov/hmrc/partnershipidentificationfrontend/service/ValidationOrchestrationService.scala:58]: Done

**Review by robertom-binqua — APPROVED** [review]

**Pull request merged**